pkg/controller/elasticsearch/user/predefined.go

```diff
@@ -204,10 +215,14 @@ func reuseOrGeneratePassword(c k8s.Client, users users, secretRef types.Namespac
 	}
 	// either reuse the password or generate a new one
 	for i, u := range users {
-		if password, exists := secret.Data[u.Name]; exists {
+		if password, exists := secret.Data[u.Name]; exists && len(password) >= params.Length {
```

Comment:
As discussed this leads to inconsistent behaviour where changing one parameter of the password policy triggers a rotation but others don't. I think I would prefer to handle rotation of credentials as an orthogonal concern for the time being and point to our existing documentation https://www.elastic.co/docs/deploy-manage/users-roles/cluster-or-deployment-auth/managed-credentials-eck#k8s-rotate-credentials

Reply:
Also the lenght parameter should only be respected when the user is on an enterrprise license and this check is not guarded against the license level, so it is incorrect anyway.